stack/base/Dockerfile

```diff
@@ -10,14 +10,11 @@ ARG AIIDA_VERSION
 
 # Install the shared requirements.
 COPY requirements.txt .
-RUN mamba install --yes \
-     aiida-core=${AIIDA_VERSION} \
-     --file requirements.txt \
-     && mamba clean --all -f -y && \
+RUN pip install --no-cache-dir aiida-core==${AIIDA_VERSION} \
+     -r requirements.txt && \
```

Comment:
This also means that things in `requirement.txt` will be installed by pip. Is that what you wanted?

Reply:
You are right. To be honest, I have no idea why this `requirements.txt` is needed, it has `aiida-core` and `pip` included, but aiida-core is already installed with specifying the version.

Reply:
It is used below as a contraint file for conda and pip. But it is indeed strange that aiida-core version is specified in two places. This has already been noted in #311. Perhaps `requirements.txt` needs to be generated dynamically in the Dockerfile?

cc @csadorf

Reply:
Indeed, the `requirements.txt` file serves a double function as constraints file. The aiida-core version is not pinned using this file, but only constraint, as you can see here: https://github.com/aiidalab/aiidalab-docker-stack/blob/a70d72b4b4be9d4d70ee842d6ad741d5bf319e7f/stack/base/requirements.txt#L1
It would need to be updated if and only if the major version is changed.

The only other requirement in that file is the pinned pip version.